core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala

```diff
@@ -1037,237 +923,6 @@ class DynamicBrokerReconfigurationTest extends QuorumTestHarness with SaslSetup
-      val updatedProps = adminZkClient.fetchEntityConfig(ConfigType.BROKER, brokerId.toString)
-      passwordConfigs.foreach { case (name, value) => assertNotEquals(props.get(value), updatedProps.get(name)) }
-
-      server.startup()
-      TestUtils.retry(10000) {
-        val newProps = adminZkClient.fetchEntityConfig(ConfigType.BROKER, brokerId.toString)
-        passwordConfigs.foreach { case (name, value) =>
-          assertEquals(passwordDecoder.decode(value), passwordDecoder.decode(newProps.getProperty(name))) }
-      }
-    }
-
-    verifyListener(SecurityProtocol.SSL, None, "add-ssl-listener-group2")
-    createAdminClient(SecurityProtocol.SSL, SecureInternal)
-    verifyRemoveListener("SSL", SecurityProtocol.SSL, Seq.empty)
-  }
-
-  @Test
-  def testAddRemoveSaslListeners(): Unit = {
-    createScramCredentials(adminClients.head, JaasTestUtils.KAFKA_SCRAM_USER, JaasTestUtils.KAFKA_SCRAM_PASSWORD)
-    createScramCredentials(adminClients.head, JaasTestUtils.KAFKA_SCRAM_ADMIN, JaasTestUtils.KAFKA_SCRAM_ADMIN_PASSWORD)
-    initializeKerberos()
-    // make sure each server's credential cache has all the created credentials
-    // (check after initializing Kerberos to minimize delays)
-    List(JaasTestUtils.KAFKA_SCRAM_USER, JaasTestUtils.KAFKA_SCRAM_ADMIN).foreach { scramUser =>
-      servers.foreach { server =>
-        ScramMechanism.values().filter(_ != ScramMechanism.UNKNOWN).foreach(mechanism =>
-          TestUtils.waitUntilTrue(() => server.credentialProvider.credentialCache.cache(
-            mechanism.mechanismName(), classOf[ScramCredential]).get(scramUser) != null,
-            s"$mechanism credentials not created for $scramUser"))
-      }}
-
-    //verifyAddListener("SASL_SSL", SecurityProtocol.SASL_SSL, Seq("SCRAM-SHA-512", "SCRAM-SHA-256", "PLAIN"))
-    verifyAddListener("SASL_PLAINTEXT", SecurityProtocol.SASL_PLAINTEXT, Seq("GSSAPI"))
-    //verifyRemoveListener("SASL_SSL", SecurityProtocol.SASL_SSL, Seq("SCRAM-SHA-512", "SCRAM-SHA-256", "PLAIN"))
-    verifyRemoveListener("SASL_PLAINTEXT", SecurityProtocol.SASL_PLAINTEXT, Seq("GSSAPI"))
-
-    // Verify that a listener added to a subset of servers doesn't cause any issues
-    // when metadata is processed by the client.
-    addListener(servers.tail, "SCRAM_LISTENER", SecurityProtocol.SASL_PLAINTEXT, Seq("SCRAM-SHA-256"))
-    val bootstrap = TestUtils.bootstrapServers(servers.tail, new ListenerName("SCRAM_LISTENER"))
-    val producer = ProducerBuilder().bootstrapServers(bootstrap)
-      .securityProtocol(SecurityProtocol.SASL_PLAINTEXT)
-      .saslMechanism("SCRAM-SHA-256")
-      .maxRetries(1000)
-      .build()
-    val partitions = producer.partitionsFor(topic).asScala
-    assertEquals(0, partitions.count(p => p.leader != null && p.leader.id == servers.head.config.brokerId))
-    assertTrue(partitions.exists(_.leader == null), "Did not find partitions with no leader")
-  }
-
-  @ParameterizedTest
-  @ValueSource(strings = Array("zk", "kraft"))
```

Comment:
Apologies, but why are we removing the entire test case instead of keeping the KRaft-related parts?